tools/mason/MasonExternal.chpl

```diff
@@ -0,0 +1,431 @@
+  select option {
+      when "--arch" do printArch();
+      when "--help" do masonExternalInfoHelp();
+      when "-h" do masonExternalInfoHelp();
+      otherwise {
+        var status = runSpackCommand("spack info " + option);
+      }
+    }
+}
+
+/* Print system arch info */
+private proc printArch() {
+  const command = "spack arch";
+  const status = runSpackCommand(command);
+}
+
+
+/* Queries system to see if package is installed on system */
+proc spkgInstalled(spec: string) {
+  const command = "spack find -df " + spec;
+  const pkgInfo = getSpackResult(command, quiet=true);
+  var found = false;
+  var dependencies: [1..0] string; // a list of pkg dependencies
+  for item in pkgInfo.split() {  
+    if item == spec {
+      return true;
+    }
+  }
+  return false;
+}
+
+
+/* Entry point into the various compiler functions */
+private proc compiler(args: [?d] string) {
+  var option = "list";
+  if args.size > 3 {
+    option = args[3];
+  }
+  select option {
+      when "--list" do listCompilers();
+      when "--find" do findCompilers();
+      when "--edit" do editCompilers();
+      otherwise do masonCompilerHelp();
+    }
+}
+
+/* lists available compilers on system */
+private proc listCompilers() {
+  const command = "spack compilers";
+  const status = runSpackCommand(command);
+ }
+
+/* Finds available compilers */
+private proc findCompilers() {
+  const command = "spack compiler find";
+  const status = runSpackCommand(command);
+}
+
+/* Opens the compiler configuration file in $EDITOR */
+private proc editCompilers() {
+  const command = "spack config edit compilers";
+  const status = runSpackCommand(command);
+}
+
+
+/* Given a toml of external dependencies returns
+   the dependencies in a toml */
+proc getExternalPackages(exDeps: unmanaged Toml) {
+
+  var exDom: domain(string);
+  var exDepTree: [exDom] unmanaged Toml;
+
+  for (name, spec) in zip(exDeps.D, exDeps.A) {
+    try! {
+      select spec.tag {
+          when fieldToml do continue;
+          otherwise {
+            var dependencies = getSpkgDependencies(spec.s);
+            const pkgInfo = getSpkgInfo(spec.s, dependencies);
+            exDepTree[name] = pkgInfo;
+          }
+        }
+    }
+    catch e: MasonError {
+      writeln(e.message());
+      exit(1);
+    }
+  }
+  return exDepTree;
+}
+
+/* Retrieves build information for MasonUpdate */
+proc getSpkgInfo(spec: string, dependencies: [?d] string) : unmanaged Toml throws {
+
+  // put above try b/c compiler comlains about return value
+  var depList: [1..0] unmanaged Toml;
+  var spkgDom: domain(string);
+  var spkgToml: [spkgDom] unmanaged Toml;
+  var spkgInfo: unmanaged Toml = spkgToml;
+
+  try {
+
+    // TODO: create a parser for returning what the user has inputted in
+    // terms of name, version, compiler etc..
+    var split = spec.split("@");
+    var pkgName = split[1];
+    var versplit = split[2].split("%");
+    var version = versplit[1];
+    var compiler = versplit[2];
```

Comment:
I wish we could infer this from `CHPL_TARGET_COMPILER` rather than force users to write it.